### ISSUE TITLE
cannot create kubernetes PKI: the certificate is not valid yet

### ISSUE THREAD

**Issue by trisberg** [docker/for-win#2913]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: EA1559AC-FFC7-4F01-B35E-17AB2F986A49/20181119163546

### Expected behavior

docker for desktop starts

### Actual behavior

it fails to start with error message:

```
time="2018-11-19T16:35:46-05:00" level=error msg="cannot create kubernetes PKI: the certificate is not valid yet"

   at Docker.Core.Pipe.NamedPipeClient.Send(String action, Object[] parameters) in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeClient.cs:line 36
   at Docker.Actions.DoStart(SynchronizationContext syncCtx, Boolean showWelcomeWindow, Boolean executeAfterStartCleanup) in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 92
   at Docker.Actions.<>c__DisplayClass19_0.<Start>b__0() in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 74
   at Docker.WPF.TaskQueue.<>c__DisplayClass19_0.<.ctor>b__1() in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.WPF\TaskQueue.cs:line 59
```

### Information

  - Windows Version: 10 Pro version 1803 (OS Build 17134.407)
  - Docker for Windows Version: Docker Desktop 2.0.0.0-win78 (28905)

### Steps to reproduce the behavior

  1. I installed Docker and enabled Kubernetes - it worked initially
  2. I ran a system update of the machine and rebooted
  3. Docker would not start after this 
      - tried uninstalling and re-installing, same error, 
      - tried rest to factory defaults - same error

**Comment by trisberg** [comment]
So, I waited a day and it all seems to be working okay today. The "the certificate is not valid yet" sounds suspicious.

**Comment by superice119** [comment]
I got this error too, and docker can't work any more.

**Comment by dgroebner** [comment]
Same here
crash report id:
AD9D7972-3FC8-42DE-AA24-DA91EC20A125/20190107164017

**Comment by ChanJerry** [comment]
Docker for Windows : Version 2.0.0.0-win81 (29211)
This happens in my machine as well.

How it happens for my machine :
1) Windows date and time, is in the Future.
2) Docker install

How to prevent:
1) Set the date and time of the Windows machine correctly 
2) Docker install

**Comment by mbitsnbites** [comment]
This happens to me too (and it seems to be related to inconsistent date/time settings). It seems very odd to me that Kubernetes needs to be involved at all to start the Docker damon, especially as Kubernetes is disabled in the Docker settings.

**Comment by erikcartman** [comment]
How can i troubleshoot this? What is correct values for date and time?

**Comment by ikelca** [comment]
same issue here
win10

**Comment by aaronz** [comment]
Delete folder C:\ProgramData\DockerDesktop and reinstall docker resolved my problem.

**Comment by mikeparker** [comment]
We're using a kubernetes function to generate the certificates which uses the exact current system clock time as a start time. If there is any system clock correction or change backwards, this will stop working. I've raised the issue on their side to see what the appetite for change is and will take it from there.

**Comment by trisberg** [comment]
Thanks for the feedback. That seems plausible and I noticed that after waiting a bit things started working.

**Comment by opsb** [comment]
Had the same issue when my macbook pro booted up with the incorrect time. Once the OS X clock updated it started fine.

**Comment by njbmartin** [comment]
Ran into this very same issue. Here's a quick summary...

For me, I had just installed some new hardware and installed a fresh copy of windows. One of the first things I did was install Docker, but ran into an issue with hardware virtualisation not being enabled yet. Before I enabled this, I noticed Windows was showing an incorrect time, despite the settings showing "set automatically" as enabled. Toggling this off and back on set the time correctly. After enabling hardware virtualisation, Docker than ran into the issue above.

After finding the issue posted here, I tried resetting Docker to factory default, which didn't work.

What fixed it was uninstalling and reinstalling Docker.
